sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -598,7 +598,9 @@ object SimplifyConditionals extends Rule[LogicalPlan] {
         // a branch with a true condition eliminates all following branches,
         // these branches can be pruned away
         val (h, t) = branches.span(_._1 != TrueLiteral)
-        CaseWhen( h :+ t.head, None)
+        // Use the value of TrueLiteral expression as elseValue to maintain
+        // nullability of CaseWhen expressions
+        CaseWhen(h :+ t.head, t.head._2)
```

Comment:
shall we update `CaseWhen#nullable` to consider conditions that are true literal? My point is we should keep the expression tree small for the good of optimizer, and the else value will never be used here.

Reply:
Got it. Yes, it is a good idea to change null check instead. I'll investigate this path.